interface/forms/eye_mag/SpectacleRx.php

```diff
@@ -43,40 +41,80 @@
 
 $form_name = "Eye Form";
 $form_folder = "eye_mag";
-require_once("php/".$form_folder."_functions.php");
+require_once("php/" . $form_folder . "_functions.php");
 
-$query = "SELECT * FROM patient_data where pid=?";
-$pat_data =  sqlQuery($query, array($data['pid']));
+//echo "<pre>";var_dump($_REQUEST);exit;
+if (!$_REQUEST['pid'] && $_REQUEST['id']) {
+    $_REQUEST['pid'] = $_REQUEST['id'];
+}
+if (!$_REQUEST['pid']) {
+    $_REQUEST['pid'] = $_SESSION['pid'];
+}
 
-$providerID  =  getProviderIdOfEncounter($encounter);
-$providerNAME = getProviderName($providerID);
-$query = "SELECT * FROM users where id = ?";
-$prov_data =  sqlQuery($query, array($providerID));
+$query = "select  *,form_encounter.date as encounter_date
+               from forms,form_encounter,form_eye_base,
+                form_eye_hpi,form_eye_ros,form_eye_vitals,
+                form_eye_acuity,form_eye_refraction,form_eye_biometrics,
+                form_eye_external, form_eye_antseg,form_eye_postseg,
+                form_eye_neuro,form_eye_locking
+                    where
+                    forms.deleted != '1'  and
+                    forms.formdir='eye_mag' and
+                    forms.encounter=form_encounter.encounter  and
+                    forms.form_id=form_eye_base.id and
+                    forms.form_id=form_eye_hpi.id and
+                    forms.form_id=form_eye_ros.id and
+                    forms.form_id=form_eye_vitals.id and
+                    forms.form_id=form_eye_acuity.id and
+                    forms.form_id=form_eye_refraction.id and
+                    forms.form_id=form_eye_biometrics.id and
+                    forms.form_id=form_eye_external.id and
+                    forms.form_id=form_eye_antseg.id and
+                    forms.form_id=form_eye_postseg.id and
+                    forms.form_id=form_eye_neuro.id and
+                    forms.form_id=form_eye_locking.id and
+                    forms.encounter=? and 
+                    forms.pid=? ";
+
+$data = sqlQuery($query, array($_REQUEST['encounter'], $_REQUEST['pid']));
+$data['ODMPDD'] = $data['ODPDMeasured'];
+$data['OSMPDD'] = $data['OSPDMeasured'];
+$data['BPDD']   = $data['ODMPDD'] + $data['OSMPDD'];
+@extract($data);
+
+$ODMPDD     = $ODPDMeasured;
+$OSMPDD     = $OSPDMeasured;
+$BPDD       = $ODMPDD + $OSMPDD;
+
+$query      = "SELECT * FROM users where id = ?";
+$prov_data  = sqlQuery($query, array($data['provider_id']));
+
+$query      = "SELECT * FROM patient_data where pid=?";
+$pat_data   = sqlQuery($query, array($data['pid']));
 
 $practice_data = $facilityService->getPrimaryBusinessEntity();
 
-if (!$_REQUEST['pid']) {
-    $_REQUEST['pid'] = $_REQUEST['id'];
-}
+$dated      = new DateTime($data['encounter_date']);
+$dated      = $dated->format('Y/m/d');
+$visit_date = oeFormatShortDate($dated);
```

Comment:
This looks really odd to me. What are you trying to do with the above block of code?

Reply:
Display it in the format the end user wants in globals.

Reply:
rather than above 2 lines, can't you just do:
```
$visit_date = oeFormatShortDate($data['encounter_date']);
```